src/server/app.py

```diff
@@ -1,11 +1,48 @@
-from flask import Flask
+import json
+
+from flask import Flask, request, Response
 from flask_cors import CORS
 
+import src.server.cea_608_encoder.byte_pair_generator as encoder
+
+
 app = Flask(__name__)
 CORS(app)
 
 
 @app.route("/submit", methods=['POST'])
 def submit():
-    pass
+    if request.get_json() is None:
+        app.logger.error('Received request with a mimetype other than JSON')
+        return Response(json.dumps({'Error': 'The endpoint requires a JSON mimetype: \'application/json\''}),
+                        status=415,
+                        mimetype='application/json')
+
+    caption_data = request.get_json()
+
+    try:
+        with open('data.json', 'w', encoding='utf-8') as file:
+            json.dump(caption_data, file, ensure_ascii=False, indent=4)
+        encoder.consume(caption_data)
+        return Response(json.dumps({'Success': 'ok'}),
+                        status=200,
+                        mimetype='application/json')
+    except IOError as err:
+        app.logger.error(f'Could not write JSON to file: {err}')
+        return Response(json.dumps({'Error': 'There was a problem trying to write the caption data to file'}),
+                        status=500,
+                        mimetype='application/json')
+    except ValueError as err:
+        app.logger.error(f'Could not encode caption data to byte pairs, shutting down with error: {err}')
+        return Response(json.dumps({'Error': 'Received bad input for one or more values.'}),
+                        status=500,
+                        mimetype='application/json')
+    except Exception:
+        return Response(json.dumps({'Error': 'Could not create byte pairs for the specified caption data, try again'}),
```

Comment:
Your ticket says that doing
`curl -d '{"Hi":"'I'm", "Writing":"To a File"}' -H "Content-Type: application/json" -X POST http://localhost:{port_number}/submit`
Should result in a 201, it results in this error instead. Here is the feedback from the server
`127.0.0.1 - - [21/Oct/2019 18:11:24] "POST /submit HTTP/1.1" 500 -
{"Error": "Could not create byte pairs for the specified caption data, try again"}`

Reply:
This is inevitable since we are looking to send the payload to the encoder. You should be able to comment out the line that consumes the caption data and it should then produce this output.

The other error codes have been updated